.github/workflows/build-publish.yml

```
@@ -0,0 +1,88 @@
+name: Build
+on: 
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NUGET_FEED: https://api.nuget.org/v3/index.json
+  NUGET_KEY: ${{ secrets.NUGET_API_KEY }}
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '9.0.300'
+
+      - name: Build
+        run: dotnet build dotnet-project-file-analyzers.slnx -c Release
+
+      - name: Test
+        run: dotnet test dotnet-project-file-analyzers.slnx -c Release --no-build --filter TestCategory!=Generators
+
+      - name: DotNet Pack
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          arrTag=(${GITHUB_REF//\// })
+          VERSION="${arrTag[2]}"
+          echo Version: $VERSION
+          VERSION="${VERSION//v}"
+          echo Clean Version: $VERSION
+          dotnet pack dotnet-project-file-analyzers.slnx -v normal --no-build --no-restore -p:PackageVersion=$VERSION -c Release --output packages
```

```suggestion
          dotnet pack dotnet-project-file-analyzers.slnx -v normal --no-build --no-restore -p:Version=$VERSION -c Release --output packages
```

In this case I think I would be in favor of the command overwriting the `Version` property, rather than the `PackageVersion` (the default of `PackageVersion` is using the value of `Version`).

---

I can agree with this